src/librustdoc/json/conversions.rs

```diff
@@ -185,6 +185,7 @@ impl From<clean::ItemKind> for ItemEnum {
             PrimitiveItem(_) | KeywordItem(_) => {
                 panic!("{:?} is not supported for JSON output", item)
             }
+            _ => unreachable!(),
```

Comment:
I think this should be explicit so this fails to compile if more variants are added or patterns are missing:

```suggestion
            ExternCrate(..) => unreachable!(),
```

Reply:
I don't think this should be here at all, see my comment above.

Reply:
I think you mean 'below' ;)